android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -472,9 +480,15 @@ public void walletInteractionDetected(WebContents webContents) {
         BraveToolbarLayoutImpl layout = getBraveToolbarLayout();
         if (layout != null) {
             layout.showWalletIcon(true);
+            calculateWalletBadgeVisibility();
         }
     }
 
+    public void calculateWalletBadgeVisibility() {
```

Comment:
why `public`?

Reply:
I was previously calling this method from different classes, Now its only called inside BraveActivity

Reply:
change to `private`

Reply:
yes, changed it